mono/utils/mono-threads-coop.c

```diff
@@ -276,64 +285,33 @@ mono_threads_init_platform (void)
 }
 
 void
-mono_threads_platform_free (MonoThreadInfo *info)
+mono_threads_coop_begin_global_suspend (void)
 {
-#ifdef TARGET_MACH
-	mach_port_deallocate (current_task (), info->native_handle);
-#endif
-
-	//See the above for what's wrong here.
-}
-
-void
-mono_threads_platform_register (MonoThreadInfo *info)
-{
-#ifdef TARGET_MACH
```

Comment:
This must come back or you'll suffer a LOT of pain later when dealing with device crashes.

The thread state dump mono-threads produce we fail to suspend requires this to allow it to be matched with an Apple crash dump. Otherwise you'll have two thread dumps with no way to connect them.